renderdoc/driver/gl/gl_pixelhistory.cpp

```diff
@@ -276,13 +383,226 @@ void QueryFailedTests(WrappedOpenGL *driver, GLPixelHistoryResources &resources,
+    else
+    {
+      driver->glReadPixels(x, y, 1, 1, eGL_RGBA, eGL_FLOAT, (void *)pixelValue.floatValue.data());
+      driver->glReadPixels(x, y, 1, 1, eGL_DEPTH_COMPONENT, eGL_FLOAT, (void *)&modValue.depth);
+
+      modValue.col = pixelValue;
+
+      if(queryType == PerFragmentQueryType::ShaderOut)
+      {
+        historyIndex->shaderOut = modValue;
+      }
+      else if(queryType == PerFragmentQueryType::PostMod)
+      {
+        historyIndex->postMod = modValue;
+      }
+    }
+    ++historyIndex;
+  }
+}
+
+void WriteClearValues(WrappedOpenGL *driver, GLPixelHistoryResources &resources,
+                      const EventUsage clearEvent, int x, int y, rdcarray<PixelModification> &history)
+{
+  driver->glBindFramebuffer(eGL_FRAMEBUFFER, resources.frameBuffer);
+  driver->glClear(eGL_COLOR_BUFFER_BIT | eGL_DEPTH_BUFFER_BIT | eGL_STENCIL_BUFFER_BIT);
+  driver->ReplayLog(0, clearEvent.eventId - 1, eReplay_WithoutDraw);
+
+  GLint savedReadFramebuffer, savedDrawFramebuffer;
+  driver->glGetIntegerv(eGL_DRAW_FRAMEBUFFER_BINDING, &savedDrawFramebuffer);
+  driver->glGetIntegerv(eGL_READ_FRAMEBUFFER_BINDING, &savedReadFramebuffer);
+  // bind our own framebuffer to save the pixel values
+  driver->glBindFramebuffer(eGL_FRAMEBUFFER, resources.frameBuffer);
+  driver->ReplayLog(clearEvent.eventId, clearEvent.eventId, eReplay_Full);
+
+  // read the post mod pixel value into the history event
+  ModificationValue modValue;
+  PixelValue pixelValue;
+  driver->glReadPixels(x, y, 1, 1, eGL_RGBA, eGL_FLOAT, (void *)pixelValue.floatValue.data());
+  driver->glReadPixels(x, y, 1, 1, eGL_DEPTH_COMPONENT, eGL_FLOAT, (void *)&modValue.depth);
+  driver->glReadPixels(x, y, 1, 1, eGL_STENCIL_INDEX, eGL_INT, (void *)&modValue.stencil);
+  modValue.col = pixelValue;
+
+  PixelModification referenceHistory;
+  referenceHistory.eventId = clearEvent.eventId;
+  auto historyIndex =
+      std::lower_bound(history.begin(), history.end(), referenceHistory,
+                       [](const PixelModification &h1, const PixelModification &h2) -> bool {
+                         return h1.eventId < h2.eventId;
+                       });
+
+  historyIndex->postMod = modValue;
+
+  // restore the capture's framebuffer
+  driver->glBindFramebuffer(eGL_DRAW_FRAMEBUFFER, savedDrawFramebuffer);
+  driver->glBindFramebuffer(eGL_READ_FRAMEBUFFER, savedReadFramebuffer);
+}
+
+bool depthTestPassed(int depthFunc, float shaderOutputDepth, float depthInBuffer)
+{
+  switch(depthFunc)
+  {
+    case eGL_NEVER: return false;
+    case eGL_LESS: return shaderOutputDepth < depthInBuffer;
+    case eGL_EQUAL: return shaderOutputDepth == depthInBuffer;
+    case eGL_LEQUAL: return shaderOutputDepth <= depthInBuffer;
+    case eGL_GREATER: return shaderOutputDepth > depthInBuffer;
+    case eGL_NOTEQUAL: return shaderOutputDepth != depthInBuffer;
+    case eGL_GEQUAL: return shaderOutputDepth >= depthInBuffer;
+    case eGL_ALWAYS: return true;
+
+    default: RDCASSERT(0);
```

Comment:
You should never assert on false/0. If there is an error condition please add an `RDCERR` with a meaningful error message like `unexpected depth function`.